homeassistant/components/weather/ipma.py

```diff
@@ -0,0 +1,177 @@
+
+    async_add_devices([IPMAWeather(station, config)], True)
+
+
+class IPMAWeather(WeatherEntity):
+    """Representation of a weather condition."""
+
+    def __init__(self, station, config):
+        """Initialise the platform with a data instance and station name."""
+        self._stationname = config.get(CONF_NAME, station.local)
+        self._station = station
+        self._condition = None
+        self._forecast = None
+        self._description = None
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    async def async_update(self):
+        """Update Condition and Forecast."""
+        with async_timeout.timeout(10, loop=self.hass.loop):
+            self._condition = await self._station.observation()
+            self._forecast = await self._station.forecast()
+            self._description = self._forecast[0].description
+
+    @property
+    def attribution(self):
+        """Return the attribution."""
+        return ATTRIBUTION
+
+    @property
+    def name(self):
+        """Return the name of the station."""
+        return self._stationname
+
+    @property
+    def condition(self):
+        """Return the current condition."""
+        return [k for k, v in CONDITION_CLASSES.items()
+                if self._forecast[0].idWeatherType in v][0]
+
+    @property
+    def temperature(self):
+        """Return the current temperature."""
+        return self._condition.temperature
+
+    @property
+    def pressure(self):
+        """Return the current pressure."""
+        return self._condition.pressure
+
+    @property
+    def humidity(self):
+        """Return the name of the sensor."""
+        return self._condition.humidity
+
+    @property
+    def visibility(self):
+        """Return the current visibility."""
+        return None
+
+    @property
+    def wind_speed(self):
+        """Return the current windspeed."""
+        return self._condition.windspeed
+
+    @property
+    def wind_bearing(self):
+        """Return the current wind bearing (degrees)."""
+        return self._condition.winddirection
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def forecast(self):
+        """Return the forecast array."""
+        if self._forecast:
+            fcdata_out = []
+            for data_in in self._forecast:
+                data_out = {}
+                data_out[ATTR_FORECAST_TIME] = data_in.forecastDate
+                data_out[ATTR_FORECAST_CONDITION] =\
+                    [k for k, v in CONDITION_CLASSES.items()
+                     if int(data_in.idWeatherType) in v][0]
+                data_out[ATTR_FORECAST_TEMP_LOW] = data_in.tMin
+                data_out[ATTR_FORECAST_TEMP] = data_in.tMax
+                data_out[ATTR_FORECAST_PRECIPITATION] = data_in.precipitaProb
+
+                fcdata_out.append(data_out)
+
+            return fcdata_out
+
+    @property
+    def state_attributes(self):
```

Comment:
Don't overwrite `state_attributes`. Use `device_state_attributes`. It will be merged with `state_attributes` during a state update.